terraform/bigquery.tf

```diff
@@ -180,6 +180,105 @@ resource "google_bigquery_table_iam_member" "event_viewers" {
   member     = each.value
 }
 
+# Events table with clustering and partitioning / IAM
+resource "google_bigquery_table" "raw_events_table" {
+  project = data.google_project.default.project_id
+
+  deletion_protection = true
+  table_id            = var.raw_events_table_id
+  dataset_id          = google_bigquery_dataset.default.dataset_id
+  schema = jsonencode([
+    {
+      "name" : "delivery_id",
+      "type" : "STRING",
+      "mode" : "NULLABLE",
+      "description" : "GUID from the GitHub webhook header (X-GitHub-Delivery)"
+    },
+    {
+      "name" : "signature",
+      "type" : "STRING",
+      "mode" : "NULLABLE",
+      "description" : "Signature from the GitHub webhook header (X-Hub-Signature-256)"
+    },
+    {
+      "name" : "received",
+      "type" : "TIMESTAMP",
+      "mode" : "NULLABLE",
+      "description" : "Timestamp for when an event is received"
+    },
+    {
+      "name" : "event",
+      "type" : "STRING",
+      "mode" : "NULLABLE",
+      "description" : "Event type from GitHub webhook header (X-GitHub-Event)"
+    },
+    {
+      "name" : "payload",
+      "type" : "JSON",
+      "mode" : "NULLABLE",
+      "description" : "Event payload JSON"
+    }
+  ])
+
+  time_partitioning {
+    field = "received" # TODO: would we rather extract an actual time from the payload?
```

Comment:
I don't really understand this in the context of TODO.  Can you rephrase it to something more imperative?

http://go/todo-style

Reply:
That was a note to myself I forgot to remove. The received time is when GMA gets the webhook from GH, not when the event was actually created. Ideally, we would parse that value from the payload and use it instead.

I think I'll put that off for now and make an enhancement issue in the GH repo.

Reply:
https://github.com/abcxyz/github-metrics-aggregator/issues/139